.github/workflows/main.yml

```diff
@@ -18,6 +18,8 @@ jobs:
   validate:
```

Comment:
can you add `permissions:` at the top of file too? If someone ever adds a new `job` and forget the permissions, it will default to `all` permissions.